src/index.js

```diff
@@ -1,9 +1,11 @@
 import assign from 'object-assign';
 import pascalCase from 'pascal-case';
+import React from 'react';
```

Comment:
This is an interesting pattern, where you *actually expect* it might fail resolving at build time… it doesn't seem like the right approach. It seems like you'd definitely want `peerDependencies` if you're going to do stuff like this… but I'd encourage avoiding `peerDependencies` too.